src/bebb/main.mo

```diff
@@ -631,6 +650,44 @@ actor {
     };
   };
 
+  /**
+   * Helper functions to match filters against all Entities
+  */
+  private func matchEntities(caller : Principal, filterCriteria : [Entity.EntityFilterCriterion]) : ?[Entity.Entity] {
+    if (filterCriteria.size() < 1) {
+      return null;
+    };
+    var foundMatches : Iter.Iter<Entity.Entity> = entitiesStorage.vals();
+    for (filterCriterion in filterCriteria.vals()) {
+      foundMatches := filterEntities(caller, filterCriterion, foundMatches)
+    };
+    return ?Iter.toArray<Entity.Entity>(foundMatches);
+  };
+
+  private func filterEntities(caller : Principal, filterCriterion : Entity.EntityFilterCriterion, entitiesToFilter : Iter.Iter<Entity.Entity>) : Iter.Iter<Entity.Entity> {
+    let buffer = Buffer.Buffer<Entity.Entity>(8);
```

Comment:
Why 8?

Reply:
Max 8 entities can be returned?

Reply:
8 is arbitrary. The buffer will grow in size if needed, so if there are more than 8 matches, it should be able to capture them all.